src/main/java/frc/robot/components/DevilDrive.java

```diff
@@ -103,12 +103,40 @@ public WheelSpeeds(double frontLeft, double frontRight, double rearLeft, double
    * @param rearLeftMotor The motor on the rear-left corner.
    * @param frontRightMotor The motor on the front-right corner.
    * @param rearRightMotor The motor on the rear-right corner.
+   * @param m_deadband
    */
   public DevilDrive(
-      SpeedController frontLeftMotor,
-      SpeedController rearLeftMotor,
-      SpeedController frontRightMotor,
-      SpeedController rearRightMotor) {
+    CANSparkMax frontLeftMotor,
+    CANSparkMax rearLeftMotor,
+    CANSparkMax frontRightMotor,
+    CANSparkMax rearRightMotor,
+    double deadband) {
+
+    requireNonNull(frontLeftMotor, "Front-left motor cannot be null");
+    requireNonNull(rearLeftMotor, "Rear-left motor cannot be null");
+    requireNonNull(frontRightMotor, "Front-right motor cannot be null");
+    requireNonNull(rearRightMotor, "Rear-right motor cannot be null");
+
+    m_frontLeftMotor = frontLeftMotor;
+    m_rearLeftMotor = rearLeftMotor;
+    m_frontRightMotor = frontRightMotor;
+    m_rearRightMotor = rearRightMotor;
+    m_deadband = deadband;
+
+    SendableRegistry.addChild(this, m_frontLeftMotor);
+    SendableRegistry.addChild(this, m_rearLeftMotor);
+    SendableRegistry.addChild(this, m_frontRightMotor);
+    SendableRegistry.addChild(this, m_rearRightMotor);
+    instances++;
+    SendableRegistry.addLW(this, "MecanumDrive", instances);
+  }
+
+  public DevilDrive(
+    CANSparkMax frontLeftMotor,
+    CANSparkMax rearLeftMotor,
+    CANSparkMax frontRightMotor,
+    CANSparkMax rearRightMotor) {
```

Comment:
I'm a little surprised to see changes in this file, since it started out identical to the WPILib MecanumDrive and then we changed the motors to velocity mode. Should these new methods be in the Chassis instead?